packages/SwingSet/src/kernel/kernel.js

```diff
@@ -808,6 +808,10 @@ export default function buildKernel(kernelEndowments) {
       const vatKeeper = kernelKeeper.allocateVatKeeperIfNeeded(vatID);
       if (!vatKeeper.isDead()) {
         vatKeeper.markAsDead();
+        const err = makeError('vat terminated');
+        for (const kpid of kernelKeeper.findPromisesFromDeadVat(vatID)) {
```

Comment:
given our brief confusion today about `p+NN / p-NN` and the difference between "owned/allocated by" and "decided by", maybe we should name this function `findPromisesDecidedByDeadVat` or just `findPromisesDecidedByVat` (since the finding function doesn't really care whether the vat is alive or dead)

Reply:
Oh also: the order in which these promises get rejected is visible to vats, so we need to make sure it's deterministic and reasonably well-specified. What do you think about accumulating the `kpid`s into a list and sorting it before sending the rejections? Another option is to require the kernelKeeper function to do the sort internally.

One downside, the most sensible sort order is numeric (`kp1, kp2, kp10`), but the easiest to implement would be lexicographic (`kp1, kp10, kp2`), ugh.

Reply:
That rename sounds sensible.  The lexicographic ordering is already guaranteed by the database, but a comment to the effect that we are relying on it won't hurt.